pymatgen/analysis/diffraction/tem.py

```diff
@@ -0,0 +1,520 @@
+# coding: utf-8
+# Copyright (c) Pymatgen Development Team.
+# Distributed under the terms of the MIT License.
+
+"""
+This module implements a TEM pattern calculator.
+"""
+
+from __future__ import division, print_function, unicode_literals
+import json
+import os
+from collections import namedtuple
+from fractions import Fraction
+from typing import List, Dict, Tuple
+import numpy as np  # type: ignore
+import scipy.constants as sc  # type: ignore
+import pandas as pd  # type: ignore
+import plotly.graph_objs as go  # type: ignore
+import plotly.offline as poff  # type: ignore
+from IPython.display import set_matplotlib_formats  # type: ignore
+from pymatgen import Structure  # type: ignore
+from pymatgen.analysis.diffraction.core import AbstractDiffractionPatternCalculator  # type: ignore
+
+with open(os.path.join(os.path.dirname(__file__),
+                       "atomic_scattering_params.json")) as f:
+    ATOMIC_SCATTERING_PARAMS = json.load(f)
+set_matplotlib_formats('retina')
+poff.init_notebook_mode(connected=True)
+# coding: utf-8
+# Copyright (c) Pymatgen Development Team.
+# Distributed under the terms of the MIT License.
+# Credit to Dr. Shyue Ping Ong for the template of the calculator
+
+__author__ = "Frank Wan, Jason L"
+__copyright__ = "Copyright 2018, The Materials Project"
+__version__ = "0.2"
+__maintainer__ = "Frank Wan respect for S.P.O"
+__email__ = "fwan@berkeley.edu, yhljason@berkeley.edu"
+__date__ = "06/19/2019, updated 01/2020"
+
+
+class TEMCalculator(AbstractDiffractionPatternCalculator):
+    """
+    Computes the TEM pattern of a crystal structure for multiple Laue zones.
+    """
+
+    def __init__(self, wavelength_cache: Dict[float, float] = {}, symprec: float = None, voltage: float = 200,
```

Comment:
`wavelength_cache` not in docstring, dangerous {} default

Reply:
Fixed.